File: src/main/java/commands/slashadapters/adapters/AnimeReleasesAdapter.java
package commands.slashadapters.adapters;

import commands.runnables.externalcategory.AnimeReleasesCommand;
import commands.slashadapters.Slash;
import commands.slashadapters.SlashAdapter;
import commands.slashadapters.SlashMeta;
import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
import net.dv8tion.jda.api.interactions.commands.OptionType;
import net.dv8tion.jda.api.interactions.commands.build.CommandData;

@Slash(command = AnimeReleasesCommand.class)
public class AnimeReleasesAdapter extends SlashAdapter {

    public CommandData addOptions(CommandData commandData) {
        return commandData
                .addOption(OptionType.STRING, "anime_name", "The name of the currently airing anime", false);
    }

    @Override
    public SlashMeta process(SlashCommandEvent event) {
        return new SlashMeta(AnimeReleasesCommand.class, collectArgs(event));
    }

}

File: src/main/java/commands/slashadapters/adapters/AnimeReleasesAdapter.kt
package commands.slashadapters.adapters

import commands.runnables.externalcategory.AnimeReleasesCommand
import commands.slashadapters.Slash
import commands.slashadapters.SlashAdapter
import commands.slashadapters.SlashMeta
import mysql.hibernate.entity.guild.GuildEntity
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.interactions.commands.OptionType
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData

@Slash(command = AnimeReleasesCommand::class)
class AnimeReleasesAdapter : SlashAdapter() {

    public override fun addOptions(commandData: SlashCommandData): SlashCommandData {
        return commandData
            .addOptions(
                generateOptionData(OptionType.STRING, "anime_name", "crunchyroll_animename", false),
                generateOptionData(OptionType.STRING, "crunchyroll_url", "crunchyroll_animeurl", false)
            )
    }

    override fun process(event: SlashCommandInteractionEvent, guildEntity: GuildEntity): SlashMeta {
        return SlashMeta(AnimeReleasesCommand::class.java, collectArgs(event))
    }

}